### PR TITLE
[5.0] Null-coalecing operator

### PR DESCRIPTION
**Pull request by Denitz**
### Summary of Changes

Use the modern and nice operator instead of isset()

### Testing Instructions

Apply patch

### Actual result BEFORE applying this Pull Request

Old-style code

### Expected result AFTER applying this Pull Request

Modern-style code

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_guidedtours/src/Model/StepModel.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ public function save($data)
 
         // Make sure we use the correct extension when editing an existing tour
         $key = $table->getKeyName();
-        $pk  = isset($data[$key]) ? $data[$key] : (int) $this->getState($this->getName() . '.id');
+        $pk  = $data[$key] ?? (int)$this->getState($this->getName() . '.id');
```
Comment (Quy) [inline administrator/components/com_guidedtours/src/Model/StepModel.php]:
```suggestion
        $pk  = $data[$key] ?? (int) $this->getState($this->getName() . '.id');
```

File: libraries/src/HTML/Helpers/Select.php [hunk 2]
```diff
@@ -644,7 +644,7 @@ public static function radiolist(
 
             $k  = $obj->$optKey;
             $t  = $translate ? Text::_($obj->$optText) : $obj->$optText;
-            $id = (isset($obj->id) ? $obj->id : null);
+            $id = ($obj->id ?? null);
```
Comment (Quy) [inline libraries/src/HTML/Helpers/Select.php]:
```suggestion
            $id = $obj->id ?? null;
```

File: libraries/src/MVC/Model/WorkflowBehaviorTrait.php [hunk 3]
```diff
@@ -394,7 +394,7 @@ protected function getStageForNewItem(Form $form, $data)
             return false;
         }
 
-        $catId = isset(((object) $data)->$catKey) ? ((object) $data)->$catKey : $form->getValue($catKey);
+        $catId = ((object)$data)->$catKey ?? $form->getValue($catKey);
```
Comment (Quy) [inline libraries/src/MVC/Model/WorkflowBehaviorTrait.php:397]:
```suggestion
        $catId = (object) $data)->$catKey ?? $form->getValue($catKey);
```

Comment (Quy) [inline libraries/src/MVC/Model/WorkflowBehaviorTrait.php]:
```suggestion
        $catId = ((object) $data)->$catKey ?? $form->getValue($catKey);
```

File: administrator/components/com_menus/src/Model/ItemModel.php [hunk 4]
```diff
@@ -1315,7 +1311,7 @@ public function rebuild()
      */
     public function save($data)
     {
-        $pk      = isset($data['id']) ? $data['id'] : (int) $this->getState('item.id');
+        $pk      = $data['id'] ?? (int)$this->getState('item.id');
```
Comment (Quy) [inline administrator/components/com_menus/src/Model/ItemModel.php]:
```suggestion
        $pk      = $data['id'] ?? (int) $this->getState('item.id');
```

File: libraries/src/HTML/Helpers/Bootstrap.php [hunk 5]
```diff
@@ -144,7 +144,7 @@ public static function carousel($selector = '', $params = []): void
             $opt = [
                 'interval' => (isset($params['interval']) ? (int) $params['interval'] : 5000),
                 'keyboard' => (isset($params['keyboard']) ? (bool) $params['keyboard'] : true),
```
Comment (Quy) [inline libraries/src/HTML/Helpers/Bootstrap.php]: Change these.

File: libraries/src/HTML/Helpers/Bootstrap.php [hunk 6]
```diff
@@ -189,7 +189,7 @@ public static function collapse($selector = '', $params = []): void
         if ($selector !== '') {
             // Setup options object
             $opt           = [];
-            $opt['parent'] = isset($params['parent']) ? $params['parent'] : false;
+            $opt['parent'] = $params['parent'] ?? false;
             $opt['toggle'] = isset($params['toggle']) ? (bool) $params['toggle'] : true;
```
Comment (Quy) [inline libraries/src/HTML/Helpers/Bootstrap.php]: Change.

**Comment by Denitz** [comment]
@Quy All done.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 13b8eae731bb3f67f29c2e4ae06a32a7e99120ec<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41657">issues.joomla.org/tracker/joomla-cms/41657</a>.</sub>

**Comment by HLeithner** [comment]
Thanks, merged by code review

**Pull request merged**